### DISCUSSION TITLE
Sunton ESP32-8048S070 board

### DISCUSSION THREAD

**Discussion by MMlodzinski** [lovyan03/LovyanGFX#424] [Device Configurations]
**Carefully written requests are more likely to be given priority.**
**丁寧に記述された要望は優先して対応される可能性が高くなります。**

### Device Name (デバイスの名称・型番等)
Sunton ESP32-8048S070
Screen driver: EK9716
Touch driver: GT911

### URL of Device Specifications document (仕様書等のURL)
[zip with documentation and Arduino demo](http://pan.jczn1688.com/directlink/1/ESP32%20module/7.0inch_ESP32-8048S070.zip)

### URL of the store where we can purchase (商品を購入できるURL)
[Aliexpress](https://pl.aliexpress.com/item/1005004952726089.html?pdp_npi=3%40dis%21PLN%21115%2C02%20z%C5%82%21114%2C94%20z%C5%82%21%21%21%21%21%40211b5e2216885581205886728ea40d%2112000031135431788%21sh01%21PL%210&spm=a2g0o.store_pc_home.productList_2005205734546.subject_3)

Is there any chance that handling this device is possible in the way that it is done with Arduino_GFX library? There it is done like this:

```cpp
#include <Arduino_GFX_Library.h>

#define GFX_BL DF_GFX_BL // default backlight pin, you may replace DF_GFX_BL to actual backlight pin
#define TFT_BL 2
/* More dev device declaration: https://github.com/moononournation/Arduino_GFX/wiki/Dev-Device-Declaration */
#if defined(DISPLAY_DEV_KIT)
Arduino_GFX *gfx = create_default_Arduino_GFX();
#else /* !defined(DISPLAY_DEV_KIT) */

/* More data bus class: https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class */
//Arduino_DataBus *bus = create_default_Arduino_DataBus();

/* More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class */
//Arduino_GFX *gfx = new Arduino_ILI9341(bus, DF_GFX_RST, 0 /* rotation */, false /* IPS */);

Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
    GFX_NOT_DEFINED /* CS */, GFX_NOT_DEFINED /* SCK */, GFX_NOT_DEFINED /* SDA */,
    41 /* DE */, 40 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
    14 /* R0 */, 21 /* R1 */, 47 /* R2 */, 48 /* R3 */, 45 /* R4 */,
    9 /* G0 */, 46 /* G1 */, 3 /* G2 */, 8 /* G3 */, 16 /* G4 */, 1 /* G5 */,
    15 /* B0 */, 7 /* B1 */, 6 /* B2 */, 5 /* B3 */, 4 /* B4 */
);
// option 1:
// 7寸 50PIN 800*480
Arduino_RPi_DPI_RGBPanel *gfx = new Arduino_RPi_DPI_RGBPanel(
  bus,
//  800 /* width */, 0 /* hsync_polarity */, 8/* hsync_front_porch */, 2 /* hsync_pulse_width */, 43/* hsync_back_porch */,
//  480 /* height */, 0 /* vsync_polarity */, 8 /* vsync_front_porch */, 2/* vsync_pulse_width */, 12 /* vsync_back_porch */,
//  1 /* pclk_active_neg */, 16000000 /* prefer_speed */, true /* auto_flush */);

    800 /* width */, 0 /* hsync_polarity */, 210 /* hsync_front_porch */, 30 /* hsync_pulse_width */, 16 /* hsync_back_porch */,
    480 /* height */, 0 /* vsync_polarity */, 22 /* vsync_front_porch */, 13 /* vsync_pulse_width */, 10 /* vsync_back_porch */,
    1 /* pclk_active_neg */, 16000000 /* prefer_speed */, true /* auto_flush */);
  
#endif /* !defined(DISPLAY_DEV_KIT) */
/*******************************************************************************
 * End of Arduino_GFX setting
 ******************************************************************************/
```

**Comment by MMlodzinski** [comment]
Hello,

I've written some lvgl example for this board using esp-idf libraries and components. Link to the repo:
[Sunton_ESP32-8048S070c LVGL example](https://github.com/MMlodzinski/Sunton_ESP32-8048S070c_ESP_IDF_LVGL_example)

As I've achieved what I wanted, fell free to close this issue. Anyway I still think that adding this board here is a good idea, so I won't close it myself.

**Comment by ChrisAnd1998** [comment]
LVGL 9.2.2 Arduino framework PlatformIO project
https://github.com/ChrisAnd1998/ESP32-8048S070C-LVGL-9
